src/main/webapp/app/entities/feature/feature.reducer.ts

```diff
@@ -0,0 +1,164 @@
+  switch (action.type) {
+    case REQUEST(ACTION_TYPES.FETCH_FEATURE_LIST):
+    case REQUEST(ACTION_TYPES.FETCH_FEATURE):
+      return {
+        ...state,
+        errorMessage: null,
+        updateSuccess: false,
+        loading: true,
+      };
+    case REQUEST(ACTION_TYPES.CREATE_FEATURE):
+    case REQUEST(ACTION_TYPES.UPDATE_FEATURE):
+    case REQUEST(ACTION_TYPES.DELETE_FEATURE):
+      return {
+        ...state,
+        errorMessage: null,
+        updateSuccess: false,
+        updating: true,
+      };
+    case FAILURE(ACTION_TYPES.FETCH_FEATURE_LIST):
+    case FAILURE(ACTION_TYPES.FETCH_FEATURE):
+    case FAILURE(ACTION_TYPES.CREATE_FEATURE):
+    case FAILURE(ACTION_TYPES.UPDATE_FEATURE):
+    case FAILURE(ACTION_TYPES.DELETE_FEATURE):
+      return {
+        ...state,
+        loading: false,
+        updating: false,
+        updateSuccess: false,
+        errorMessage: action.payload,
+      };
+    case SUCCESS(ACTION_TYPES.FETCH_FEATURE_LIST):
+      return {
+        ...state,
+        loading: false,
+        entities: action.payload.data,
+      };
+    case SUCCESS(ACTION_TYPES.FETCH_FEATURE):
+      return {
+        ...state,
+        loading: false,
+        entity: action.payload.data,
+      };
+    case SUCCESS(ACTION_TYPES.CREATE_FEATURE):
+    case SUCCESS(ACTION_TYPES.UPDATE_FEATURE):
+      return {
+        ...state,
+        updating: false,
+        updateSuccess: true,
+        selectedFeature: null,
+        entity: action.payload.data,
+      };
+    case SUCCESS(ACTION_TYPES.DELETE_FEATURE):
+      return {
+        ...state,
+        updating: false,
+        updateSuccess: true,
+        selectedFeature: null,
+        entity: {},
+      };
+    case ACTION_TYPES.RESET:
+      return {
+        ...initialState,
+        selectedFeature: null,
+      };
+    case ACTION_TYPES.SELECT_FEATURE:
+      return {
+        ...initialState,
+        selectedFeature: action.payload,
+      };
+    default:
+      return state;
+  }
+};
+
+const apiUrl = 'api/features';
+
+// Actions
+
+export const getViewFeaturesEntities: ICrudGetViewFeaturesAction<IFeature> = viewId => ({
+  type: ACTION_TYPES.FETCH_FEATURE_LIST,
+  payload: axios.get<IFeature>(`${apiUrl}?view=${viewId}&cacheBuster=${new Date().getTime()}`),
+});
+
+export const getEntities: ICrudGetAllAction<IFeature> = (page, size, sort) => ({
+  type: ACTION_TYPES.FETCH_FEATURE_LIST,
+  payload: axios.get<IFeature>(`${apiUrl}?cacheBuster=${new Date().getTime()}`),
+});
+
+export const getEntity: ICrudGetAction<IFeature> = id => {
+  const requestUrl = `${apiUrl}/${id}`;
+  return {
+    type: ACTION_TYPES.FETCH_FEATURE,
+    payload: axios.get<IFeature>(requestUrl),
+  };
+};
+
+export const createEntity: ICrudPutAction<IFeature> = entity => async dispatch => {
+  const result = await dispatch({
+    type: ACTION_TYPES.CREATE_FEATURE,
+    payload: axios.post(apiUrl, cleanEntity(entity)),
+  });
+  dispatch(getEntities());
```

Comment:
this should be done when request is successful. wait for the success in container and then call getEntities

Reply:
its auto generated file by jhipster

Reply:
remove dispatch(getEntities());